motion_adaption_node:

  wait_for_tf_in_sec:             0.5
  world_ref_frame_name:           /openni_depth_frame
  user:
    torso_frame_name:             /torso_filtered
    head_frame_name:              /head_filtered
    right_shoulder_frame_name:    /right_shoulder_filtered
    right_elbow_frame_name:       /right_elbow_filtered
    right_hand_frame_name:        /right_hand_filtered
    left_shoulder_frame_name:     /left_shoulder_filtered
    left_elbow_frame_name:        /left_elbow_filtered
    left_hand_frame_name:         /left_hand_filtered
  robot:
    base_frame_name:              /base_link
    torso_ref_frame_name:         /torso_1_link

    head_frame_name:              /head_2_link    
    right_shoulder_frame_name:    /arm_right_2_link
    right_elbow_frame_name:       /arm_right_4_link
    right_hand_frame_name:         /arm_right_7_link   
    left_shoulder_frame_name:     /arm_left_2_link
    
  # define roations of orientations by roll pitch yaw
  rotation_angles:
    # align the orientation of the user torso with robot torso
    # i.e.: user bends foward -> ref frame rotates forward
    ref_frame_rot: {p: 0.0, r: 1.5707963267948966, y: -1.5707963267948966}  
    # align adapted frames with goal frames
    goals:
      torso_goal_rot: {p: 1.5707963267948966, r: 3.1415926535897931, y: 0.0}
      head_goal_rot: {p: 1.5707963267948966, r: 3.1415926535897931, y: 0.0}
      right_elbow_goal_rot: {p: 0.0, r: 3.1415926535897931, y: 3.1415926535897931}
      right_hand_goal_rot: {p: 1.5707963267948966, r: 0.0, y: 0.0}
      left_elbow_goal_rot: {p: 3.1415926535897931, r: 0.0, y: 0.0}
      left_hand_goal_rot: {p: -1.5707963267948966, r: 0.0, y: 0.0}

#----------------------------------------------------------------------------------------------------------------------#

tree_kinematics_node:

  urdf_xml_model:       robot_description

  nr_of_endpoints:      6
  endpoint_0:           torso_2_link
  endpoint_1:           head_2_link
  endpoint_2:           arm_right_4_link
  endpoint_3:           hand_right_link
  endpoint_4:           arm_left_4_link
  endpoint_5:           hand_left_link

  # frequency of the service calls for IK calculations in hz
  # used to calculate velocities and to adapt the maximum velocties specified below
  srv_call_frequency: 10
  
  # the precision of the endeffectors position the IK solver tries to reach
  epsilon: 0.010
  
  # the maximum number of iterations the IK solver will execute
  max_iterations: 10

  # Factor applied to the maximum velocities specified in the urdf model of the robot in rad/sec
  q_dot_max_factor: 1.0
  q_dot_min_factor: 0.3
  
  low_pass_factor:  1.0

  # minimum and maximum translational and rotational velocities of the endeffectors in m/sec
  x_dot_trans_min: 0.2
  x_dot_trans_max: 0.2
  x_dot_rot_min: 0.0628
  x_dot_rot_max: 0.0682
    
  # Factor to prevent running through kinematic singularities
  lambda: 0.01

  js_w_matr/j0:  0.0
  js_w_matr/j1:  0.0
  js_w_matr/j2:  0.0
  js_w_matr/j3:  0.0
  js_w_matr/j4:  1.0
  js_w_matr/j5:  1.0
  js_w_matr/j6:  1.0
  js_w_matr/j7:  1.0
  js_w_matr/j8:  1.0
  js_w_matr/j9:  1.0
  js_w_matr/j10: 1.0
  js_w_matr/j11: 1.0
  js_w_matr/j12: 1.0
  js_w_matr/j13: 1.0
  js_w_matr/j14: 1.0
  js_w_matr/j15: 1.0
  js_w_matr/j16: 1.0
  js_w_matr/j17: 1.0
  js_w_matr/j18: 1.0
  js_w_matr/j19: 1.0
  js_w_matr/j20: 1.0
  js_w_matr/j21: 1.0
  js_w_matr/j22: 1.0
  js_w_matr/j23: 1.0

  ts_w_matr/endpt0/px: 0.2
  ts_w_matr/endpt0/py: 0.2
  ts_w_matr/endpt0/pz: 0.2
  ts_w_matr/endpt0/rx: 0.2
  ts_w_matr/endpt0/ry: 0.2
  ts_w_matr/endpt0/rz: 0.2

  ts_w_matr/endpt1/px: 1.0
  ts_w_matr/endpt1/py: 1.0
  ts_w_matr/endpt1/pz: 1.0
  ts_w_matr/endpt1/rx: 1.0
  ts_w_matr/endpt1/ry: 1.0
  ts_w_matr/endpt1/rz: 1.0

  ts_w_matr/endpt2/px: 0.2
  ts_w_matr/endpt2/py: 0.2
  ts_w_matr/endpt2/pz: 0.2
  ts_w_matr/endpt2/rx: 0.2
  ts_w_matr/endpt2/ry: 0.2
  ts_w_matr/endpt2/rz: 0.2

  ts_w_matr/endpt3/px: 1.0
  ts_w_matr/endpt3/py: 1.0
  ts_w_matr/endpt3/pz: 1.0
  ts_w_matr/endpt3/rx: 1.0
  ts_w_matr/endpt3/ry: 1.0
  ts_w_matr/endpt3/rz: 1.0

  ts_w_matr/endpt4/px: 0.0
  ts_w_matr/endpt4/py: 0.0
  ts_w_matr/endpt4/pz: 0.0
  ts_w_matr/endpt4/rx: 0.0
  ts_w_matr/endpt4/ry: 1.0
  ts_w_matr/endpt4/rz: 1.0

  ts_w_matr/endpt5/px: 0.0
  ts_w_matr/endpt5/py: 0.0
  ts_w_matr/endpt5/pz: 0.0
  ts_w_matr/endpt5/rx: 0.0
  ts_w_matr/endpt5/ry: 1.0
  ts_w_matr/endpt5/rz: 1.0
  
#----------------------------------------------------------------------------------------------------------------------#

reem_teleop_coordinator_node:

  nr_of_endpoints:      6
  endpoint_0:           torso_2_link
  endpoint_1:           head_2_link
  endpoint_2:           arm_right_4_link
  endpoint_3:           hand_right_link
  endpoint_4:           arm_left_4_link
  endpoint_5:           hand_left_link
  endpoint_0_goal:      torso_goal
  endpoint_1_goal:      head_goal
  endpoint_2_goal:      r_elbow_goal
  endpoint_3_goal:      r_hand_goal
  endpoint_4_goal:      l_elbow_goal
  endpoint_5_goal:      l_hand_goal
  root_frame_name:      ref_frame
  
  check_self_collision: true
  check_joint_limits:   true  

  loop_rate: 2

